### PR TITLE
fix: redis instrumentation loses context when using callbacks

### PR DESCRIPTION
**Pull request by aspectom**
## Which problem is this PR solving?

- When using redis instrumentation with callbacks instead of async await, original span context is lost.

## Short description of the changes
- Call context.with when calling original callback

### CODE REVIEW

File: plugins/node/opentelemetry-instrumentation-redis/src/utils.ts [hunk 1]
```diff
@@ -146,7 +147,10 @@ export const getTracedInternalSendCommand = (
         }
 
         endSpan(span, err);
-        return originalCallback.apply(this, arguments);
+        const callbackThis = this;
+        return context.with(originalContext, () => {
+          return originalCallback.apply(callbackThis, arguments);
+        });
```
Comment (dyladan) [inline plugins/node/opentelemetry-instrumentation-redis/src/utils.ts:153]: You can also just do `return context.with(originalContext, originalCallback, callbackThis, ...arguments)` https://open-telemetry.github.io/opentelemetry-js-api/classes/contextapi.html#with

**Review by blumamir — APPROVED** [review]

File: plugins/node/opentelemetry-instrumentation-redis/src/utils.ts (hunk 1 above)
Comment (dyladan) [reply plugins/node/opentelemetry-instrumentation-redis/src/utils.ts:153]: This way you avoid creating an additional function call

Comment (aspectom) [reply plugins/node/opentelemetry-instrumentation-redis/src/utils.ts]: Wasn't aware of that one. thanks @dyladan , fixed :)

**Review by MSNev — APPROVED** [review]

File: plugins/node/opentelemetry-instrumentation-redis/src/utils.ts [hunk 2]
```diff
@@ -125,6 +125,7 @@ export const getTracedInternalSendCommand = (
       );
     }
 
+    const originalContext = context.active();
```
Comment (Flarna) [inline plugins/node/opentelemetry-instrumentation-redis/src/utils.ts:128]: You could move this line down into the `if (originalCallback) {`.

**Review by Flarna — APPROVED** [review]

File: plugins/node/opentelemetry-instrumentation-redis/src/utils.ts (hunk 2 above)
Comment (aspectom) [reply plugins/node/opentelemetry-instrumentation-redis/src/utils.ts]: Thanks @Flarna, fixed.

**Review by vmarchaud — APPROVED** [review]

**Review by Rauno56 — APPROVED** [review]

**Review by obecny — APPROVED** [review]

**Pull request merged**